### PR TITLE
Hybrid vs Northstar notifications

### PR DESCRIPTION
**Pull request by NISHIY-EKSDEE**
### Jira Ticket
https://adc.luxoft.com/jira/browse/STVCIS-2327

### Purpose
Add feature to publish HybridVsNorthstar report URL to Hybrid PR

### Effect of change
Updated commit message in HybridVsNorthStar branch (now contains Hybrid PR URL)

### :warning: Notes for Reviewers
Reports won't be published until the GithubNotificator is initialized

### Jenkins Builds
https://rpr.cis.luxoft.com/job/DevJobs/job/DevHybridVsNorthstarAuto/job/hybrid_auto_PR-855/11/
https://rpr.cis.luxoft.com/job/DevJobs/job/DevHybridVsNorthstarAuto/job/hybrid_auto_PR-855/15/
## Test Hybrid PR
https://github.com/Radeon-Pro/RPRHybrid/pull/855
## Test HybridVsNorthstar branch
https://github.com/Radeon-Pro/HybridVsNorthStar/tree/hybrid_auto_PR-855

### CODE REVIEW

**Review by Speedwag00n — APPROVED** [review]

File: vars/hybrid_vs_northstar_pipeline.groovy [hunk 1]
```diff
@@ -67,10 +67,13 @@ def createHybridBranch(Map options) {
         if (options.comparisionBranch) {
             dir("HybridVsNorthstar") {
                checkoutScm(branchName: "main", repositoryUrl: PROJECT_REPO)
-
+               String message = "Triggered by Build #${env.BUILD_NUMBER}"
+               if (env.CHANGE_URL) {
+                   message += ". Pull URL - ${env.CHANGE_URL}"
```
Comment (epozine) [inline vars/hybrid_vs_northstar_pipeline.groovy:72]:
```suggestion
                   message += ". PR URL - ${env.CHANGE_URL}"
```

File: vars/hybrid_vs_northstar_pipeline.groovy [hunk 2]
```diff
@@ -347,8 +350,13 @@ def executePreBuild(Map options)
             stash includes: "Northstar64.dll", name: "Northstar64Dll", allowEmpty: false
 
             if (env.BRANCH_NAME && env.BRANCH_NAME.startsWith("hybrid_auto_")) {
-
-                options.hybridBuildNumber = options.commitMessage.replace("Triggered by Build #", "") as Integer
+                if (options.commitMessage.contains("Pull URL")) {
```
Comment (epozine) [inline vars/hybrid_vs_northstar_pipeline.groovy:353]:
```suggestion
                if (options.commitMessage.contains("PR URL")) {
```

File: vars/hybrid_vs_northstar_pipeline.groovy [hunk 3]
```diff
@@ -347,8 +350,13 @@ def executePreBuild(Map options)
...
+                    def parts = options.commitMessage.replace("Triggered by Build #", "").replace("Pull URL - ", "").split(". ")
```
Comment (epozine) [inline vars/hybrid_vs_northstar_pipeline.groovy:354]:
```suggestion
                    def parts = options.commitMessage.replace("Triggered by Build #", "").replace("PR URL - ", "").split(". ")
```

**Review by epozine — APPROVED** [review]
Мне кажется Pull URL лучше заменить на PR URL. Это минор, остальное все хорошо

File: vars/hybrid_vs_northstar_pipeline.groovy [hunk 4]
```diff
@@ -524,6 +532,10 @@ def executeDeploy(Map options, List platformList, List testResultList) {
                     GithubNotificator.updateStatus("Deploy", "Building test report", "success", options, NotificationConfiguration.REPORT_PUBLISHED, "${BUILD_URL}/Test_20Report")
                 }
             }
+            GithubNotificator githubNotificator = new GithubNotificator(this, options)
+            githubNotificator.init(options)
+            options["githubNotificator"] = githubNotificator
+            GithubNotificator.sendPullRequestComment(options.hybridPullUrl, "Hybrid vs Northstar report - ${BUILD_URL}/Test_20Report", options)
```
Comment (epozine) [inline vars/hybrid_vs_northstar_pipeline.groovy:538]:
```suggestion
            GithubNotificator.sendPullRequestComment(options.hybridPullUrl, "Hybrid vs Northstar comparison report - ${BUILD_URL}/Test_20Report", options)
```

**Review by epozine — APPROVED** [review]

**Pull request merged**